src/app/meshes/sources.ts

```diff
@@ -1,18 +1,51 @@
 import { DataSourceResponse } from '@/app/application/services/data-source/DataSourcePool'
 import type KumaApi from '@/services/kuma-api/KumaApi'
 import type {
+  PaginatedApiListResponse as CollectionResponse,
+} from '@/types/api.d'
+import type {
+  Mesh,
   MeshInsight,
 } from '@/types/index.d'
 
 type MeshParams = {
   mesh: string
 }
+type PaginationParams = {
+  size: number
+  page: number
+}
+
+type Closeable = {close: () => void}
+
+export type MeshSource = DataSourceResponse<Mesh>
+export type MeshCollection = CollectionResponse<Mesh>
+export type MeshCollectionSource = DataSourceResponse<MeshCollection>
+
 export type MeshInsightSource = DataSourceResponse<MeshInsight>
+
 export const sources = (api: KumaApi) => {
   return {
-    '/:mesh/insights': async (params: MeshParams, source: {close: () => void}) => {
+    '/:mesh/meshes': async (params: MeshParams & PaginationParams, source: Closeable) => {
       source.close()
-      return api.getMeshInsights({ name: params.mesh })
+      const offset = params.size * (params.page - 1)
+      return api.getAllMeshes({
+        size: params.size,
+        offset,
+      })
     },
+    '/:mesh/mesh': (params: MeshParams, source: Closeable) => {
+      source.close()
+      return api.getMesh({ name: params.mesh })
+    },
+    '/:mesh/insights': async (params: MeshParams, source: Closeable) => {
+      source.close()
+      if (params.mesh === '*') {
```

Comment:
Why is mesh with 2 endpoints 1 for list and one for get single item and insights is not list this? Isn't this going to be confusing too because depending on the param different response types are going to be removed.

I'd be very much for having 2 sources for insights.

Reply:
I changed this up, at this moment (at least) we only need the `/:mesh/insights` source.